collections/running_reduce.ts

```diff
@@ -3,26 +3,45 @@
 
 /**
  * Calls the given reducer on each element of the given collection, passing it's
- * result as the accumulator to the next respective call, starting with the given
- * initialValue. Returns all intermediate accumulator results.
+ * result as the accumulator to the next respective call. If initialValue is given,
+ * starting with that value. Returns all intermediate accumulator results.
  *
  * Example:
  *
  * ```ts
  * import { runningReduce } from "https://deno.land/std@$STD_VERSION/collections/mod.ts";
  * import { assertEquals } from "https://deno.land/std@$STD_VERSION/testing/asserts.ts";
  *
- * const numbers = [1, 2, 3, 4, 5];
- * const sumSteps = runningReduce(numbers, (sum, current) => sum + current, 0);
+ * const firstNumbers = [1, 2, 3, 4, 5];
+ * const result = runningReduce(firstNumbers, (sum, current) => sum + current);
  *
- * assertEquals(sumSteps, [1, 3, 6, 10, 15]);
+ * assertEquals(result, [1, 3, 6, 10, 15]);
+ *
+ * const secondNumbers = [1, 2, 3, 4, 5]
+ * const resultWithInitialValue = runningReduce(secondNumbers, (sum, current) => sum + current, 5);
+ *
+ * assertEquals(resultWithInitialValue, [6, 8, 11, 15, 20]);
  * ```
  */
-export function runningReduce<T, O>(
+export function runningReduce<T>(
   array: readonly T[],
-  reducer: (accumulator: O, current: T) => O,
```

Comment:
[kotlin.collections/running-reduce](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/running-reduce.html#runningreduce) does not accept different types

```kotlin
fun main() {
    val strings = listOf(1, "foo", 2, "bar")
    println(strings.reduce { acc, current -> acc + current })
    // Type mismatch: inferred type is Unit but {Comparable<*> & java.io.Serializable} was expected
}
```